catalyst/dl/callbacks/optimizer.py

```diff
@@ -73,26 +81,38 @@ def on_epoch_start(self, state):
         for i in range(len(optimizer.param_groups)):
             safitty.set(optimizer.param_groups, i, "weight_decay", value=0.0)
 
-    def on_batch_start(self, state):
-        state.loss = None
-
-    def on_batch_end(self, state):
+    def _get_loss(self, state) -> torch.Tensor:
         loss = state.get_key(key="loss", inner_key=self.loss_key)
-        if isinstance(loss, dict):
-            loss = list(loss.values())
+
         if isinstance(loss, list):
-            loss = torch.mean(torch.stack(loss))
+            if len(loss) > 1:
+                logger.warning(
+                    f"Attention, the loss is a list. "
+                    f"Only the last value will be used for `backward`."
+                    f"To aggregate losses into "
+                    "one value use `CriterionAggregatorCallback`"
+                )
+            loss = loss[-1]
```

Comment:
why do we take last loss by default?

Reply:
Because in case of a list, the last loss is appended to the and
```
        if state.loss is not None:
            if isinstance(state.loss, list):
                state.loss.append(loss)
            else:
                state.loss = [state.loss, loss]
```

Reply:
and why only last one? I think it is much better to sum/mean them, or just throw an exception, because last-loss usage looks very tricky

Reply:
fixed